snap/patches/default_data_dir.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/util.cpp b/src/util.cpp
+index 57b0419..dd21329 100644
+--- a/src/util.cpp
++++ b/src/util.cpp
+@@ -467,7 +467,7 @@ boost::filesystem::path GetDefaultDataDir()
+     return GetSpecialFolderPath(CSIDL_APPDATA) / "faircoin2";
+ #else
+     fs::path pathRet;
+-    char* pszHome = getenv("HOME");
++    char* pszHome = getenv("SNAP_USER_COMMON");
```

Comment:
I explain it here: https://github.com/faircoin/faircoin/pull/10/files#diff-5fde7a6d86053f0e1d88c0a2a238941fR32

Basically, snaps are confined to their own directories and that's what gives them stronger security. So the faircoin snap will only be able to write, by default, to /home/user/snap/faircoin. When the binary is launched, $HOME is overwritten to point to /home/user/snap/faircoin/version1, which is ok from the security point of view.

However, when the snap is updated to version2, all the files in the version1 directory are copied. That's very cool to be able to rollback in the case there's a problem. But in the case of bitcoin, that would mean copying the full blockchain. In faircoin, the blockchain will start small, but it will eventually also grow to hundreds of gigas. So, we have a directory in ~/snap/faircoin/common that is not copied during updates.

Usually, the programs have a flag to specify where to store data. But that's not the case of bitcoin-qt. The only way I found to tell bitcoin-qt to save the blockchain by default in SNAP_USER_COMMON was to patch the source code when the snap is building.